scraper/scraper.py

```diff
@@ -0,0 +1,212 @@
+        # format query list as a proper query string
+        query = 'action=search'
+        for specifier in queryList:
+            query += '&s[]={0}'.format(specifier)
+
+        # specify CourseBook request header
+        coursebookHeader = {
+            'Cookie': 'PTGSESSID={0}'.format(self.ptgsessid),
+            'Content-Length': '46',
+            'Accept': '*/*',
+            'X-Requested_with': 'XMLHttpRequest',
+            'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.81 Safari/537.36',
+            'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
+            'Sec-Gpc': '1',
+            'Sec-Fetch-Site': 'same-origin',
+            'Sec-Fetch-Mode': 'cors',
+            'Sec-Fetch_dest': 'empty',
+            'Referer': 'https://coursebook.utdallas.edu/',
+        }
+
+        # send CourseBook search query
+        # self.logger.info('Sending CourseBook query: {0}'.format(query))
+        try:
+            response = requests.post(
+                'https://coursebook.utdallas.edu/clips/clip-cb11-hat.zog',
+                headers = coursebookHeader,
+                data = query,
+            )
+            html = BeautifulSoup(response.content, 'html.parser')
+        except:
+            self.logger.error('Failed to receive response from query: {0}'.format(query))
+            return {}
+
+        # parse response for items returned
+        try:
+            numItems = html.find_all('b')[0].__repr__().split('(')[1].split(')')[0].split(' ')[0] # this should probably be cleaned up
+            if numItems == 'no':
+                self.logger.info('0 results obtained from query: {0}'.format(query))
+                return {}
+        except:
+            self.logger.error('Failed to parse number of items returned from query: {0}'.format(query))
+            return {}
+
+        # parse response for download link
+        try:
+            downloadLink = html.find_all('a')[0]['href'].replace('\\','').replace('"','')
+            if downloadLink[0:14] == '/reportmonkey/':
+                downloadID =  downloadLink.split('/')[-1]
+            else:
+                raise Exception()
+        except:
+            self.logger.error('Failed to parse download link to {0} item(s) from query: {1}'.format(numItems, query))
+            return {}
+
+        # create reportmonkey downloadURL from downloadID
+        downloadURL = 'https://coursebook.utdallas.edu/reportmonkey/cb11-export/{0}/{0}/json'.format(downloadID)
+
+        # specify reportmonkey request header
+        reportmonkeyHeader = {
+            'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9',
+            'Accept-Encoding': 'gzip, deflate, br',
+            'Accept-Language': 'en-US,en;q=0.9',
+            'Connection': 'keep-alive',
+            'Cookie': 'PTGSESSID={}'.format(self.ptgsessid),
```

Comment:
f-strings are cooler.
```suggestion
            'Cookie': f'PTGSESSID={self.ptgsessid}',
```